pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller_test.go

```diff
@@ -0,0 +1,91 @@
+package machinehealthcheck
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"strconv"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	aroclient "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned"
+	arofake "github.com/Azure/ARO-RP/pkg/operator/clientset/versioned/fake"
+	mock_dynamichelper "github.com/Azure/ARO-RP/pkg/util/mocks/dynamichelper"
+)
+
+// Test reconcile function
+func TestReconciler(t *testing.T) {
+	log := logrus.NewEntry(logrus.StandardLogger())
+	type test struct {
+		name   string
+		arocli aroclient.Interface
+		mocks  func(mdh *mock_dynamichelper.MockInterface)
+	}
+
+	for _, tt := range []*test{
+		{
+			name: "Feature Flag disabled, no action",
+			arocli: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					OperatorFlags: arov1alpha1.OperatorFlags{
+						ENABLED: strconv.FormatBool(false),
+					},
+				},
+			}),
+			mocks: func(mdh *mock_dynamichelper.MockInterface) {
+				mdh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Times(0)
+			},
+		},
+		{
+			name: "When enabled dynamic helper ensures resources",
+			arocli: arofake.NewSimpleClientset(&arov1alpha1.Cluster{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: arov1alpha1.SingletonClusterName,
+				},
+				Spec: arov1alpha1.ClusterSpec{
+					OperatorFlags: arov1alpha1.OperatorFlags{
+						ENABLED: strconv.FormatBool(true),
+					},
+				},
+			}),
+			mocks: func(mdh *mock_dynamichelper.MockInterface) {
+				mdh.EXPECT().Ensure(gomock.Any(), gomock.Any()).Return(nil).Times(1)
+			},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			mdh := mock_dynamichelper.NewMockInterface(controller)
+
+			tt.mocks(mdh)
+
+			ctx := context.Background()
+			r := &Reconciler{
+				log:    log,
+				arocli: tt.arocli,
+				dh:     mdh,
+			}
+			request := ctrl.Request{}
+			request.Name = "cluster"
+
+			_, err := r.Reconcile(ctx, request)
+
+			if err != nil {
+				t.Error(err)
+				t.Error("----------------------")
```

Comment:
We typically use this assertion in other tests:
```
			_, err := r.Reconcile(ctx, request)
			if err != nil && err.Error() != tt.wantErr ||
				err == nil && tt.wantErr != "" {
				t.Error(err)
			}
```
Also, I would assert errors even if you expect them to be nil.